### PR TITLE
Feature/skysight

### LINKED ISSUE #837: Skysight wheater
It would be a nice feature, to get the skysight wheater as overlay for the moving map. Especially the wave parameters.

### PR DESCRIPTION
**Pull request by Scumi**
Related to #837.

This PR provides a working SkySight implementation, together with some minor alignment of the other weather providers.
Most of the credit goes to @August2111 and @CazYokoyama for the huge amount of work they put into this feature. I ported their implementation from OpenSoar to XCSoar and adapted it to XCSoar’s new weather control flow. Hopefully, I have not introduced any issues during that process. 
My own tests have been satisfactory, but the implementation still needs a thorough review - particularly for edge cases - and more eyes on the overall integration.

## Summary by CodeRabbit

* **New Features**
  * Added SkySight weather overlays with selectable layers and forecast times.
  * Added SkySight account and region settings.
  * Added automatic updates, forecast preloading, offline cache use, and cache cleanup.
  * Added support for live tiles and multi-overlay map display.
* **Bug Fixes**
  * Improved weather download retries, throttling recovery, and forecast decoding.
  * Improved handling of GeoTIFF and georeferenced image overlays.
* **Documentation**
  * Documented SkySight overlays, caching, input controls, and build requirements.
* **Tests**
  * Added coverage for SkySight overlays, forecast preloading, and busy states.

### CODE REVIEW

**Comment by Plantain** [comment]
Anyone who wants to improve, maintain or mainline this is welcome to contact me for complimentary SkySight access. I can also answer implementation questions.

**Comment by yorickreum** [comment]
> Anyone who wants to improve, maintain or mainline this is welcome to contact me for complimentary SkySight access. I can also answer implementation questions.

I'd be very interested in SkySight access to further test the feature. I didn't see your mail address here but you can reach out to me at xcsoar@yorickreum.de

**Comment by lordfolken** [comment]
@Plantain if you could reactivate my account, so i can validate it. philipp@wollschlegel.ch

File: test/src/TestWeatherOverlayPagePlacement.cpp [hunk 1]
```diff
@@ -60,6 +60,34 @@ TestReplaceCurrentOverlay()
...
+  settings.SetDefaults();
+
+  ok1(!SetSkysightLayerOnPage(settings, 0, "thermal"));
+  ok1(ApplyWeatherOverlayToPage(settings, 0,
+                                PageLayout::Overlay::SKYSIGHT,
+                                -1, "wind"));
+  ok1(SetSkysightLayerOnPage(settings, 0, "thermal"));
+  ok1(settings.pages[0].skysight_overlay == "thermal");
+}
+
```
[the remark this answers is not in the document]
Comment (Scumi) [reply test/src/TestWeatherOverlayPagePlacement.cpp:131]: 26 seems correct to me...

**Comment by Scumi** [comment]
@Plantain I'm interested as well, I also have some questions around download limits and throttling.

**Comment by CazYokoyama** [comment]
Glad to see the PR still alive. I rebased it for logical parts while I don't have full understanding of the code. Thank you.

**Comment by Plantain** [comment]
> @Plantain I'm interested as well, I also have some questions around download limits and throttling.

You will have to either send me an email (skysight@skysight.io) or give me some way to contact you.

Regarding download limits/throttling, it should not be hit in normal use.
However there remains a bug in some XCSoar implementations where it appears to go into an infinite loop hitting our API at 10-100QPS and using a tremendous amount of user mobile data, which will very quickly get you throttled/banned. I have no idea if this implementation has fixed that. I believe OpenSoar did fix it.

**Comment by Plantain** [comment]
And if I could add a project-wide nit, it should be SkySight not Skysight.

**Comment by Scumi** [comment]
> > @Plantain I'm interested as well, I also have some questions around download limits and throttling.
> 
> You will have to either send me an email ([skysight@skysight.io](mailto:skysight@skysight.io)) or give me some way to contact you.
> 
> Regarding download limits/throttling, it should not be hit in normal use. However there remains a bug in some XCSoar implementations where it appears to go into an infinite loop hitting our API at 10-100QPS and using a tremendous amount of user mobile data, which will very quickly get you throttled/banned. I have no idea if this implementation has fixed that. I believe OpenSoar did fix it.

Interesting, I will have a look at it. I'm pretty sure its still there as I see this all the time and tried to work around it. It only happens for Rain and Satellite, so that path does something weird.

**Comment by Plantain** [comment]
https://github.com/XCSoar/XCSoar/pull/1684 had some prior discussion on it, and as far as I understand @August2111 did find the problem on OpenSoar

**Comment by Plantain** [comment]
@Scumi I think I addressed all the pending issues here on skysightsoaringweather/XCSoar:feature/skysight , please cherry-pick 8568db3075 141d0a7731 to this PR, or I can force push here if you'd rather.

**Comment by Scumi** [comment]
Nice, thanks. I'll have a look. I'm half way through with the fixes as well, so I need to compare the approach.

**Comment by lordfolken** [comment]
## Rebase / follow-up notes

Rebased onto current `master` and force-pushed to this branch. Summary of what landed beyond the rebase:

### From Plantain (`skysightsoaringweather/XCSoar`)
- Cherry-picked **SkySight** branding rename (paths, symbols, profile keys, docs) plus the MinGW NetCDF `_wstat64` patch for PC/WIN32OPENGL CI.
- Cherry-picked retry / resource bounding (tile failure caps, decode cancel, etc.).

### Branding
- Treat **SkySight** as an untranslated brand name (bare `"SkySight"` for titles / empty controls; phrases like `_("SkySight Email")` still translated).
- Legacy `Skysight*` profile / page keys still load so early test profiles keep working.

### Remaining review fixes (folded into the resource-bounds commit)
- Sanitize API `region` / `layer` ids before cache filenames (`SkySight::IsSafeId`); UTF-8-validate region names.
- Arm the 1 Hz poll timer only when credentials are configured.
- OpenGL: `assert(InMainThread())` in `Render()`; non-OpenGL: early-return so the DrawThread does not touch SkySight state.

`NEWS.txt` conflict resolved; SkySight entry kept under `* ui` with `#2690`.

**Comment by lordfolken** [comment]
Tracks / implements [#837](https://github.com/XCSoar/XCSoar/issues/837) (Skysight weather). Planned for **v7.46**.

**Comment by Plantain** [comment]
I get at least a request every week for this to happen (and currently I am pointing those users to Navigator/Jet/Copilot), so would be great if this could have some attention for v7.45.

**Comment by Scumi** [comment]
I have the update ready by tomorrow.

**Comment by OBrown92** [comment]
@lordfolken do you think this will make it soon? This is one of the main reasons I still use the JET version. But JET does not have the new XCSoar Cloud functions...

**Comment by lordfolken** [comment]
> @lordfolken do you think this will make it soon? This is one of the main reasons I still use the JET version. But JET does not have the new XCSoar Cloud functions...

Can you download the apk from actions and test it? That would help.

its the last issue blocking [v7.45](https://github.com/XCSoar/XCSoar/issues?q=state%3Aopen%20milestone%3A%22v7.45%22) milestone.

**Comment by OBrown92** [comment]
Sure. Tested it on my device right now and it is working fine. Not so intuitive like in the Jet Version but with much more features. Having two different layers on two sites is nice, also the switcher. Will probably not fly within the next two weeks but will try to test it a bit in my car 😆

**Comment by August2111** [comment]
Oh dear, I suppose I’m too late now with my comments, potential improvements, and corrections... I’m sorry for only weighing in on the SkySight topic now—but unfortunately, I’ve been a bit overwhelmed with other matters over the last few weeks and months ;-(
Now I’ll have to try comparing Scumi’s differences, improvements, and potential issues with my current setup: there’s a lot I really like, but there are also things I don’t find quite as successful.
So, you’ll likely be launching with the current state of version 7.45—since it’s already been merged—and I’ll try to gradually introduce the things I consider important!
Best regard — and good luck!
August2111

**Comment by lordfolken** [comment]
@August2111 First of all: Thank you for your immense work, without you this wouldn't have been possible. 

Am i happy with the entire feature? Probably not. 
Is there bugs? Very likely.
Also for me the netcdf approach as a whole is very low resolution and inefficient in both rendering and data transfer. 
So i will suspect that this will remain an area of concern. 

The feature is sufficiently isolated, so we can work and change things, without affecting XCSoar as a whole.

Further delaying it would have yielded only marginally better results. 

Lets get it in the hand of the users, and see how it breaks. ;-)
